core/domain/exp_services.py

```diff
@@ -792,6 +793,15 @@ def _save_exploration(committer_id, exploration, commit_message, change_list):
                 'which is too old. Please reload the page and try again.'
                 % (exploration_model.version, exploration.version))
 
+    if feconf.ENABLE_STRING_CLASSIFIER:
```

Comment:
I don't think I understand the issue. AFAICT, get_trainable_states_dict() does not take in a version, so I don't see what the problem is here...